### DISCUSSION TITLE
Parallel simulation in real-time(qrt) mode and with custom tstep

### DISCUSSION THREAD

**Discussion by TheASingh** [CURENT/andes#352] [Q&A]
Hi Dr. Cui,

In example 6, I found a way to do parallel simulation. But I want to do it in real-time as well as with my custom tstep.
In example 6, tf, qrt, tstep, shrinkt are not mentioned. But I tried to pass these parameters as below:

`systems = andes.run('*.xlsx', routine='tds', pool=True, verbose=10, tf=100, no_output=True, shrinkt=0, tstep=0.1, qrt=1)`

I get the following output and it completed under 18 seconds.
```
Cases are processed in the following order:
"ieee09-Gen-PVD1-ESD1-redmva-qpri-withoutlimiter-reducePQ.xlsx"
"ieee09_Gen_PVD1_ESD1_for2hour.xlsx"
Log saved to "C:\Users\AMRIND~1\AppData\Local\Temp\2\andes-6kgjfc3l\andes.log".
-> Multiprocessing finished in 17.6642 seconds.
```
Below are my queries:
1. Is it run for 100s ?
2. As it completed in 18 seconds, Am I passing the `qrt` parameter correctly?
3. And the `tstep` I'm providing is working or not?

Please let me know is there any other way to pass `tf, qrt, tstep, shrinkt` parameters?

**Comment by TheASingh** [comment]
To be able to run parallel simulation in **real-time** and **with custom tstep**, I made the following change in the **main.py** file (line no. 404-414).


        if system.is_setup:
            system.PFlow.run(**kwargs)
            for r in routine:
                if r == 'tds':
                    if 'qrt' in kwargs:
                        system.TDS.config.qrt = kwargs['qrt']
                    if 'tstep' in kwargs:
                        system.TDS.config.tstep = kwargs['tstep']
                    if 'shrinkt' in kwargs:
                        system.TDS.config.shrinkt = kwargs['shrinkt']
                system.__dict__[routine_cli[r.lower()]].run(**kwargs)

**Reply by cuihantao** [reply]
Yeah, but what I meant is that these arguments can be passed to `config_option` without modifying the code. Namely, 

`ss = andes.run("test_case.xlsx", config_option=['TDS.qrt=1', 'TDS.tstep=0.033', 'TDS.shrinkt=0'])`

**Reply by cuihantao** [reply]
If you don't need to apply it on the fly, you can also write the new settings in the config file.

**Comment by cuihantao** [comment]
Check the examples for argument “config_option”. Additional options can be passed to it.

Regards,
Hantao Cui
